packages/bp/src/core/messaging/messaging-service.ts

```diff
@@ -56,19 +56,23 @@ export class MessagingService {
 
     this.messaging = new MessagingChannel({
       url: this.getMessagingUrl(),
-      adminKey: process.env.INTERNAL_PASSWORD,
-      axios: this.getAxiosConfig()
+      axios: this.getAxiosConfig(),
+      adminKey: this.isExternal ? process.env.MESSAGING_ADMIN_KEY : process.env.INTERNAL_PASSWORD
     })
     this.messaging.on('user', this.handleUserNewEvent.bind(this))
     this.messaging.on('started', this.handleConversationStartedEvent.bind(this))
     this.messaging.on('message', this.handleMessageNewEvent.bind(this))
 
-    await AppLifecycle.waitFor(AppLifecycleEvents.STUDIO_READY)
+    if (!this.isExternal) {
+      await AppLifecycle.waitFor(AppLifecycleEvents.STUDIO_READY)
+    }
     this.messaging.url = this.getMessagingUrl()
```

Comment:
Normally we wait for the studio to be ready as a way of waiting for the messaging server to be started. Since the messaging server isn't started by botpress when it's external, I just skip this step here

Reply:
Yes, good catch!